### DISCUSSION TITLE
How to workaround styles overlapping with third-party libraries?

### DISCUSSION THREAD

**Discussion by allomov** [activeadmin/activeadmin#7917] [Help] [answered]
### Summary

I use Material UI with React in my project and want to use them to build custom components for index grid items or forms.

For instance, I can have a form that uses `Box`, `TextField`, and `RadioGroup` MUI react components (like is described [here](https://mui.com/material-ui/react-text-field/)). But as a result, I get a form with mixed styles that looks in the following way:

<img width="387" alt="image" src="https://user-images.githubusercontent.com/122246/228621868-85845c43-fac1-4c49-b0bb-dfa79dbdb9d3.png">

### Expected behavior

Ideally, there could be a way to remove default active admin styles to apply styles from third-party libraries.

### How to reproduce

You can find an example application that shows the form from the image above here: https://github.com/allomov/active_admin_with_mui

### What I've tried?

I've tried to find a way to tell CSS to avoid applying CSS rules from active admin to a container with some CSS class. For instance, by adding the following code to `active_admin.scss`:
```scss
:not(.no-active-admin-styles) {
  @import "active_admin/mixins";
  @import "active_admin/base";
}
```
Or looking into using `all` property (see details [here](https://developer.mozilla.org/en-US/docs/Web/CSS/all )). But all this didn't work, and I finished with a workaround like this:

```css
/* Active Admin CSS */
@media screen {
    // ...
    .active_admin_form, #sidebar, .comments {
      @import "active_admin/forms";
    }
    // ...
}
```
But I do not feel like this solution is complete. I will be grateful for suggestions. Thank you!

**Comment by javierjulio** [comment]
Addressing the root issue is something we'd like to do but won't be possible without major updates. Sadly, with forms it's mostly out of our control due to our dependency on formtastic. It would be a significant project. Instead of closing this out, since you are also asking a general question on how to workaround integrating a third party front end library, I'll move this to discussions. Thank you.

**Comment by allomov** [comment, marked as answer]
In order to solve this issue I decided to copy [_base.scss](https://github.com/activeadmin/activeadmin/blob/7dd9bc8/app/assets/stylesheets/active_admin/_base.scss) file to `./vendor/assets/stylesheets/active_admin/_base.scss` and commented out the line with `@import "active_admin/components/grid";`. This looks like a work around, but it may work since it allows us to have better control over CSS styles.

**Reply by javierjulio** [reply]
@allomov you should look into the v4 beta. A month or two after you opened your initial issue I started on a major update using TailwindCSS after some research. The styles are in a much better place now as they are scoped. We do apply defaults to form inputs but all other styles are scoped and with low specificity. It should be an improvement, less conflict in CSS, with v4 now.
